### PR TITLE
Martin conversions

### PR DESCRIPTION
**Pull request by BMartinos**
This PR relates to issues: 
#349
#346
#344

### CODE REVIEW

**Comment by BMartinos** [comment]
Hey @devcritter Can you review this PR for me when you get a chance. Thanks

**Comment by BMartinos** [comment]
Hey @devcritter can you review this PR for me when you get a chance. Thanks

File: src/api/mediators.coffee [hunk 1]
```diff
@@ -5,131 +5,99 @@ logger = require 'winston'
...
+
+    if typeof existing != 'undefined' and existing != null
+      if semver.gt(mediator.version, existing.version)
         yield Mediator.findByIdAndUpdate(existing._id, mediator).exec();
-      }
-    } else {
-      if (!mediator.endpoints || mediator.endpoints.length<1) {
-        throw {
-          name: 'ValidationError',
+      else
```
Comment (devcritter) [inline src/api/mediators.coffee]: rather take out the else if there's no code following it

File: src/api/mediators.coffee [hunk 2]
```diff
@@ -5,131 +5,99 @@ logger = require 'winston'
...
-    logger.info('Mediator with urn '+urn+' has been successfully removed by %s', this.authenticated.email);
-  }catch(e){
-    logger.error('Could not remove Mediator by urn '+urn+' via the API: ' + e);
-    this.body = e.message;
-    this.status = 'internal server error';
-  }
-
-}`
\ No newline at end of file
+      else
```
Comment (devcritter) [inline src/api/mediators.coffee]: same comment as above

**Comment by devcritter** [comment]
Thanks! looks good. Just some minor comments: instead of doing
```
  if condition
    someStuff()
  else
otherStuff()
```
I think it's better to leave the extra else:
```
  if condition
    someStuff()
otherStuff()
```

**Comment by BMartinos** [comment]
Reopening in new PR with diff merge location

**Pull request closed without merging**